src/main/java/de/danielbechler/diff/selector/CollectionItemElementSelector.java

```diff
@@ -71,6 +94,6 @@ public boolean equals(final Object o)
 	@Override
 	public int hashCode()
 	{
-		return item != null ? item.hashCode() : 0;
+		return item != null ? identityStrategy.hashCode(item) : 0;
```

Comment:
Yes. Since it only applies to the lookup of element selectors of nodes that have actually changed I don't think this will have a massive impact.